ivr/src/main/java/org/motechproject/ivr/service/impl/OutboundCallServiceImpl.java

```diff
@@ -130,6 +151,32 @@ public void initiateCall(String configName, Map<String, String> parameters) {
             addCallDetailRecord(CallDetailRecord.CALL_FAILED, config, params, motechCallId);
             throw new CallInitiationException(message, e);
         }
+    }
+
+    private void generateCallFile(Config config, Map<String, String> params, Map<String, String> completeParams, String motechCallId) throws URISyntaxException, IOException {
+        String myUri = mergeUriAndRemoveParams(config.getOutgoingCallUriTemplate(), completeParams);
+
+        Path path = Paths.get(new URI(myUri));
+        Path tempDirectory = Files.createTempDirectory(path.getParent(), "motech");
+        Path tempPath = Files.createTempFile(tempDirectory, "motech", ".tmp");
+
+        BufferedWriter bufferedWriter = Files.newBufferedWriter(tempPath);
+        for (Map.Entry<String, String> entry : completeParams.entrySet()) {
+            bufferedWriter.write(String.format("%s: %s", entry.getKey(), entry.getValue()));
+            bufferedWriter.newLine();
+        }
+        bufferedWriter.close();
```

Comment:
use the writer in a try-with-resource, closing resources should be done in a finally block